tests/e2e/microservice.go

```diff
@@ -15,79 +15,73 @@ import (
 )
 
 const (
-	msImage       = "busybox"
-	msImageTag    = "1.31"
-	msStopTimeout = 3 // seconds
-	msLabelKey    = "e2e.test.ms"
-	msNamePrefix  = "e2e-test-"
+	msDefaultImage = "busybox:1.31"
+	msStopTimeout  = 1 // seconds
+	msLabelKey     = "e2e.test.ms"
+	msNamePrefix   = "e2e-test-ms-"
 )
 
 var (
 	linuxPingRegexp = regexp.MustCompile("\n([0-9]+) packets transmitted, ([0-9]+) packets received, ([0-9]+)% packet loss")
 )
 
 type microservice struct {
-	ctx          *TestCtx
-	name         string
+	ctx *TestCtx
+
+	name string
+
 	dockerClient *docker.Client
 	container    *docker.Container
 	nsCalls      nslinuxcalls.NetworkNamespaceAPI
 }
 
-type pingOpts struct {
-	allowedLoss int
-	outIface    string
-}
-
-type pingOpt func(opts *pingOpts)
-
-func createMicroservice(ctx *TestCtx, msName string, dockerClient *docker.Client, nsCalls nslinuxcalls.NetworkNamespaceAPI) *microservice {
+func createMicroservice(ctx *TestCtx, msName string, dockerClient *docker.Client, nsCalls nslinuxcalls.NetworkNamespaceAPI) (*microservice, error) {
 	msLabel := msNamePrefix + msName
-	container, err := dockerClient.CreateContainer(docker.CreateContainerOptions{
-		Name: msLabel,
+
+	opts := docker.CreateContainerOptions{
+		Context: ctx.ctx,
+		Name:    msLabel,
 		Config: &docker.Config{
-			Env:   []string{"MICROSERVICE_LABEL=" + msLabel},
-			Image: msImage + ":" + msImageTag,
-			Cmd:   []string{"tail", "-f", "/dev/null"},
+			Image: msDefaultImage,
 			Labels: map[string]string{
 				msLabelKey: msName,
 			},
+			Env: []string{"MICROSERVICE_LABEL=" + msLabel},
+			Cmd: []string{"sleep", "600"},
```

Comment:
Go test has by default timeout of 10min. I just wanted to avoid possibility that these containers somehow stay in the system after unclean exit. Ill change back now.. but maybe in the future we could do somethiing like:

```go
	cmd := []string{"tail", "-f", "/dev/null"}
	if deadline, ok := ctx.t.Deadline(); ok {
		timeoutSec := int(time.Until(deadline).Seconds())
		cmd = []string{"sleep", fmt.Sprint(timeoutSec)}
	}
```